chromium_src/net/dns/dns_client.cc

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "base/feature_list.h"
+#include "net/base/features.h"
+#include "net/dns/dns_util.h"
+#include "net/dns/public/dns_over_https_server_config.h"
+
+#include <vector>
+
+namespace net {
+namespace {
+
+struct fallbackEndpoint {
+  const char* name;
+  const char* address;
+};
+
+constexpr fallbackEndpoint endpoints[] = {
```

Comment:
pls consider:
1. adding `enum class DohFallbackEndpointType { ... };` and using it to create `DohFallbackEndpoint` struct that you then export as `base::span<const DohFallbackEndpoint> GetDohFallbackEndpoints();` via `net/dns/public/secure_dns_mode.h` header.
2. using `-1` as a default value for the feature and handle it accordingly in UI/Android codebase.

don't use strings as identifiers, don't use 0 as a default value.

Reply:
I ended up using an enum class as you suggested, which brings some helpful type checking in C++. But it does result in using strings as identifiers in Java. Interested in your thoughts on this approach.

Reply:
> But it does result in using strings as identifiers in Java. Interested in your thoughts on this approach.

IMO this sounds better than using arbitrary integers for feature parameters.